### PR TITLE
ILLUSIONS: Add keymapper support

### PR DESCRIPTION
**Pull request by NabeelShabbir**

### CODE REVIEW

File: engines/illusions/input.h [hunk 1]
```diff
@@ -26,6 +26,7 @@
 #include "common/events.h"
 #include "common/keyboard.h"
 #include "common/rect.h"
+#include "illusions.h"
```
Comment (mgerhardy) [inline engines/illusions/input.h:29]: include via `"illusions/illusions.h"` please

File: engines/illusions/metaengine.cpp [hunk 2]
```diff
@@ -75,6 +82,87 @@ void IllusionsMetaEngine::removeSaveState(const char *target, int slot) const {
...
+	act = new Action("INVENTORY", _("Open inventory"));
+	act->setCustomEngineActionEvent(kActionInventory);
+	act->addDefaultInputMapping("i");
+	act->addDefaultInputMapping("JOY_Y");
+	gameKeyMap->addAction(act);
+
+	act = new Action("ENCHEAT", _("Enable cheat mode"));
+	act->setCustomEngineActionEvent(kActionCheatMode);
+	act->addDefaultInputMapping("F1");
+	act->addDefaultInputMapping("JOY_LEFT_SHOULDER");
```
Comment (OMGPizzaGuy) [inline engines/illusions/metaengine.cpp:156]: Hmmm.. What does cheat mode do for this engine? I feel odd about accidentally cheating in a game by just trying out the buttons on the gamepad.

Comment (NabeelShabbir) [reply engines/illusions/metaengine.cpp:156]: The user has to input a certain keyword then only is he able to activate it.

Comment (OMGPizzaGuy) [reply engines/illusions/metaengine.cpp:156]: Alright - I checked that out an it makes sense. The one thing I would be interested in is a debugger console command to activate the cheat as well. That would probably fall out of scope for this PR.

**Review by OMGPizzaGuy — APPROVED** [review]

**Comment by bluegr** [comment]
Thanks!

**Pull request merged**